app/src/org/commcare/CommCareApplication.java

```diff
@@ -220,6 +223,14 @@ public void onCreate() {
             AppUtils.checkForIncompletelyUninstalledApps();
             initializeAnAppOnStartup();
         }
+
+        LocalePreferences.saveDeviceLocale(Locale.getDefault());
+    }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {

```

Comment:
Since this method is called in application class and not activity, there's no point in putting android:configChanges for any activity tag in manifest, since we're not registering configChanges in out activities. App's onConfigChanged will always be called when new lang is selected in android settings.

Reply:
👍